### PR TITLE
Shepherd update

### PR DESCRIPTION
**Pull request by obuisard**
### Summary of Changes

Added constants for step types and interactive types
Update to Shepherd 11.0.1

### CODE REVIEW

File: administrator/components/com_guidedtours/src/Extension/GuidedtoursComponent.php [hunk 1]
```diff
@@ -28,6 +28,78 @@ class GuidedtoursComponent extends MVCComponent implements BootableExtensionInte
...
     use HTMLRegistryAwareTrait;
 
+    /**
+     * The step type
+     *
+     * @since __DEPLOY_VERSION__
+     */
+    public const STEP_TYPE_NAMES = [
+        self::STEP_NEXT   => 'COM_GUIDEDTOURS_FIELD_VALUE_NEXT_STEP',
+        self::STEP_REDIRECT => 'COM_GUIDEDTOURS_FIELD_VALUE_REDIRECT',
```
Comment (brianteeman) [inline administrator/components/com_guidedtours/src/Extension/GuidedtoursComponent.php]: would it not be better for these three language keys to be
COM_GUIDEDTOURS_FIELD_VALUE_STEP_NEXT
COM_GUIDEDTOURS_FIELD_VALUE_STEP_REDIRECT
COM_GUIDEDTOURS_FIELD_VALUE_STEP_INTERACTIVE

Comment (brianteeman) [reply administrator/components/com_guidedtours/src/Extension/GuidedtoursComponent.php]: or 
COM_GUIDEDTOURS_FIELD_STEP_NEXT_VALUE

Comment (obuisard) [reply administrator/components/com_guidedtours/src/Extension/GuidedtoursComponent.php]: Yeah... I will fix that

File: administrator/components/com_guidedtours/src/Extension/GuidedtoursComponent.php [hunk 2]
```diff
@@ -28,6 +28,78 @@ class GuidedtoursComponent extends MVCComponent implements BootableExtensionInte
...
+    public const STEP_INTERACTIVE = 2;
+
+    /**
+     * The step type
+     *
+     * @since __DEPLOY_VERSION__
+     */
+    public const STEP_INTERACTIVETYPE_NAMES = [
+        self::STEP_INTERACTIVETYPE_FORM_SUBMIT   => 'COM_GUIDEDTOURS_TYPE_INTERACTIVE_STEP_FORM_SUBMIT_LABEL',
+        self::STEP_INTERACTIVETYPE_TEXT => 'COM_GUIDEDTOURS_TYPE_INTERACTIVE_STEP_TEXT_FIELD_LABEL',
```
Comment (brianteeman) [inline administrator/components/com_guidedtours/src/Extension/GuidedtoursComponent.php]: whatever the naming convention used for the keys it should be consistent

File: administrator/components/com_guidedtours/src/Extension/GuidedtoursComponent.php [hunk 3]
```diff
@@ -28,6 +28,78 @@ class GuidedtoursComponent extends MVCComponent implements BootableExtensionInte
...
+        self::STEP_INTERACTIVETYPE_BUTTON    => 'COM_GUIDEDTOURS_TYPE_INTERACTIVE_STEP_BUTTON_LABEL',
+        self::STEP_INTERACTIVETYPE_OTHER    => 'COM_GUIDEDTOURS_TYPE_INTERACTIVE_STEP_OTHER_LABEL',
+    ];
+
+    /**
+     * An interactive step where a user click on a form button.
```
Comment (brianteeman) [inline administrator/components/com_guidedtours/src/Extension/GuidedtoursComponent.php]:
```suggestion
     * An interactive step where a user clicks on a form button.
```

File: administrator/components/com_guidedtours/src/Extension/GuidedtoursComponent.php (hunk 2 above)
Comment (obuisard) [reply administrator/components/com_guidedtours/src/Extension/GuidedtoursComponent.php]: Agreed

File: administrator/components/com_guidedtours/src/Extension/GuidedtoursComponent.php (hunk 3 above)
Comment (obuisard) [reply administrator/components/com_guidedtours/src/Extension/GuidedtoursComponent.php]: Thanks!

**Pull request merged**